animation_nodes/algorithms/mesh/triangulate_mesh.pyx

```diff
@@ -0,0 +1,316 @@
+
+    cdef float angle
+    cdef Vector3 v1, v2, v3
+    cdef Vertex* verticesData
+    cdef Py_ssize_t polyStart, triangleIndex, polyIndex
+    cdef Py_ssize_t j, index, previousIndex, earIndex, nextIndex
+
+    polyIndex = 0
+    triangleIndex = 0
+    for i in range(polygonAmount):
+        polyStart = oldPolyStarts[i]
+        polyLength = oldPolyLengths[i]
+
+        # Initialization of polygon's vertices.
+        verticesData = <Vertex*>PyMem_Malloc(polyLength * sizeof(Vertex))
+        projectPolygonVertices(vertices, polyStart, polyLength, oldIndices, verticesData)
+
+        # Check polygon polarity.
+        if not polyCounterClockwise(verticesData):
+            for j in range(polyLength):
+                previousIndex = verticesData[j].previous
+                nextIndex = verticesData[j].next
+                verticesData[j].previous = nextIndex
+                verticesData[j].next = previousIndex
+
+        # Calculate angle and status of vertices.
+        for j in range(polyLength):
+            angle = calculateAngleWithSign(verticesData, j)
+            verticesData[j].angle = angle
+            if angle >= 0.0 and not pointsInTriangle(polyLength, verticesData, j):
+                verticesData[j].isEar = True
+            else:
+                verticesData[j].isEar = False
+
+        # Calculate triangle polygon indices.
+        for j in range(polyLength - 2):
+            if polyLength - 2 - j == 1:
+                newPolyStarts[triangleIndex] = polyIndex
+                newPolyLengths[triangleIndex] = 3
+                triangleIndex += 1
+
+                index = 0
+                for k in range(polyLength):
+                    if verticesData[k].isEar:
+                        newIndices[polyIndex + index] = verticesData[k].index
+                        index += 1
+                        if index == 3: break
+                polyIndex += 3
+                break
+
+            # Removing an ear which has smallest inner-angle.
+            earIndex = findEarHasMinAngle(polyLength, verticesData)
+
+            newPolyStarts[triangleIndex] = polyIndex
+            newPolyLengths[triangleIndex] = 3
+            triangleIndex += 1
+
+            newIndices[polyIndex] = verticesData[verticesData[earIndex].previous].index
+            newIndices[polyIndex + 1] = verticesData[earIndex].index
+            newIndices[polyIndex + 2] = verticesData[verticesData[earIndex].next].index
+            polyIndex += 3
+
+            removeEarVertex(polyLength, verticesData, earIndex)
+
+        PyMem_Free(verticesData)
+
+    return newPolygons
+
+# Remove ear-vertex and update neighbor-vertices.
+cdef void removeEarVertex(Py_ssize_t polyLength, Vertex* verticesData, Py_ssize_t earIndex):
+    cdef Py_ssize_t previousIndex, nextIndex
+    previousIndex = verticesData[earIndex].previous
+    nextIndex = verticesData[earIndex].next
+
+    verticesData[earIndex].isEar = False
```

Comment:
The ear is being removed, so you don't need to update any of its values. More on this later.

Reply:
`verticesData[earIndex].isEar = False` is removed.